pytests/conftest.py

```diff
@@ -0,0 +1,244 @@
+from tblib import pickling_support
+
+pickling_support.install()
+
+import contextlib
+import functools
+import multiprocessing as mp
+import os
+import posixpath
+import signal
+import socket
+import sys
+import tempfile
+from six import reraise as raise_
+from six.moves.urllib.parse import quote, urljoin, urlunsplit
+
+import pytest
+import requests
+import requests_unixsocket
+
+import bjoern
+
+
+def free_port():
+    with contextlib.closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+        s.bind(('', 0))
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        port = s.getsockname()[1]
+    return port
+
+
+class ErrorHandleMiddleware:
+    def __init__(self, app, pipe=None):
+        self.app = app
+        self.pipe = pipe
+
+    def __call__(self, *args, **kwargs):
+        try:
+            return self.app(*args, **kwargs)
+        except Exception as e:
+            if self.pipe is not None:
+                exc_info = sys.exc_info()
+                exc_type = exc_info[0]
+                tb = exc_info[2]
+                self.pipe.send((exc_type, e, tb))
+            raise e
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_call(item):
+    yield
+    if client.__name__ in item.funcargs.keys():
+        # check for uncaught errors in server process, raise if any
+        testclient = item.funcargs[client.__name__]
+        testclient._reraise_app_errors()
+
+
+def reraise(f):
+    @functools.wraps(f)
+    def wrapper(*args, **kwargs):
+        result = f(*args, **kwargs)
+        testclient = args[0]
```

Comment:
better make this the first `wrapper` argument explicitly

Reply:
Agreed!